### DISCUSSION TITLE
Response to Akka's license change

### DISCUSSION THREAD

**Discussion by henricook** [playframework/playframework#11403] [General]
I'm somewhat reeling today from reading about Akka's license change: https://alexn.org/blog/2022/09/07/akka-is-moving-away-from-open-source/

Does this mean that using Play commercially may soon involve an Akka license cost?

Can Play keep using an older (open licensed) version of Akka indefinitely?

**Comment by henricook** [comment]
Noted: Playframework exemptions in the license

https://github.com/akka/akka/blob/main/LICENSE

**Comment by tsuyoshizawa** [comment]
As stated in the [Additional Use Grant](https://github.com/akka/akka/blob/main/LICENSE#L8-L16), it reads that there is no license cost if you simply use the Play Framework functionality.

**Reply by ErunamoJAZZ** [reply]
This will mean that any directly import of akka-streams will be out of that use grant?

**Reply by BillyAutrey** [reply]
The wording in the license and blog post seem to imply (to my reading) that implicit use of Akka Streams is covered.  That is, if you use it transitively and don't call the API directly.  However, if you invoke the Akka Streams source directly, or use its API, that requires a license.

I'm not speaking as an official representative in this response.  This is just how I interpret the license and blog post.

**Comment by hertg** [comment]
Even though Play Framework is allowed to continue using Akka without a license, I think Lightbend's move is in its core so anti open-source that I'd be in favor for Play to pin the dependency to 2.6 or replace Akka with a fork that isn't affected by the license change. 

If Play Framework users use Akka outside of the scope of Play they will have to do this anyway. And if Play hangs on to Lightbend's "generous" offer, to be granted a free license, unnecessary dependency version conflicts will probably be the result.

I do not have the knowledge about Play's development to assess what effect such a decision would have. But simply from a users perspective, and as someone who values opensource very highly, there is a discussion to be had if Play, a project that has proven itself to be a viable community project after Lightbend's abandonment, really should take the free license bait and co-sign Lightbend's questionable move.

**Comment by tototoshi** [comment]
Play initially used Netty as its backend. Is it possible to make Netty the default backend again as before?

I appreciate Play exemption, but it sounds complicated and unclear. I think it would be good to reduce the implicit akka dependency as much as possible not to encourage license violations.

**Reply by ndeverge** [reply]
And would it mean to restore the `play-iteratees` project?

**Reply by y-yu** [reply]
Oleg et al are developing a new stream library: https://github.com/strymonas/strymonas 👀
(Unfortunately it works only Scala 3....)

**Comment by jducoeur** [comment]
I'm pretty sure that switching away from akka-streams entirely would be extremely disruptive to users.  But I agree that pinning to the last truly open-source version would be wise.  (Really, forking and shading that version would be ideal: anything else is going to lead to dependency hell for some users.  But that's a good deal of work.)

**Comment by mkurz** [comment]
Hey guys, just to let you know, regarding the Akka license change, we are discussing things internally. (I have COVID-19 unfortunately so a bit idle currently).

**Comment by BillyAutrey** [comment]
In an effort to be open and transparent about our position, we opted to make a small statement of our intent.  It will take time to work through all the implications, but we are doing that now.

Announcement here - https://github.com/orgs/playframework/discussions/11405
